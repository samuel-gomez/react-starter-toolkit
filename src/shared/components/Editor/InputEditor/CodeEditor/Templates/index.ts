export { default } from './Templates';
export type { Tlistelements } from './Templates';
