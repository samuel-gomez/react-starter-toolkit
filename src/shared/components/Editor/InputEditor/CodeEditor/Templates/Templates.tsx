import { Button } from '@axa-fr/react-toolkit-all';
import HelpInfo from 'shared/components/HelpInfo';
import { TReturnUseCodeEditor } from '../CodeEditor';

export const DEFAULT_LIST = [
  {
    id: 'add-list',
    icon: 'list',
    tooltipLabel: 'Liste',
  },
  {
    id: 'add-title__h1',
    label: 'H1',
    tooltipLabel: 'Titre h1',
  },
  {
    id: 'add-title__h2',
    label: 'H2',
    tooltipLabel: 'Titre h2',
  },
  {
    id: 'add-title__h3',
    label: 'H3',
    tooltipLabel: 'Titre h3',
  },
  {
    id: 'add-title__h4',
    label: 'H4',
    tooltipLabel: 'Titre h4',
  },
  {
    id: 'add-text',
    label: 'P',
    tooltipLabel: 'Paragraph',
  },
  {
    id: 'add-restitution__twoColumns',
    icon: 'list-alt',
    tooltipLabel: 'Restitution two columns',
  },
  {
    id: 'add-restitution__oneColumn',
    icon: 'list-alt',
    tooltipLabel: 'Restitution one column',
  },
  {
    id: 'add-modal',
    icon: 'modal-window',
    tooltipLabel: 'Modal',
  },
  {
    id: 'add-panel__basic',
    icon: 'credit-card',
    tooltipLabel: 'Panel basic',
  },
];

type TlistElement = {
  id: string;
  label?: string;
  icon?: string;
  tooltipLabel?: string;
};
export type Tlistelements = TlistElement[];

type TTemplates = {
  list?: Tlistelements;
  submitTemplate: TReturnUseCodeEditor['submitTemplate'];
  onClearCodeEditor: TReturnUseCodeEditor['onClearCodeEditor'];
};

const Templates = ({ list = DEFAULT_LIST, submitTemplate, onClearCodeEditor }: TTemplates) => (
  <div className="af-templates">
    <HelpInfo key="delete" content="Supprimer" classModifier="editor">
      <Button className="af-btn--circle af-btn--delete" onClick={onClearCodeEditor}>
        <i className="glyphicon glyphicon-trash"></i>
      </Button>
    </HelpInfo>
    {list.map(({ id, label, icon, tooltipLabel }) => (
      <HelpInfo key={id} content={tooltipLabel} classModifier="editor">
        <Button className="af-btn--circle" id={id} onClick={submitTemplate}>
          {!!icon && <i className={`glyphicon glyphicon-${icon}`}></i>}
          {label}
        </Button>
      </HelpInfo>
    ))}
  </div>
);

export default Templates;
