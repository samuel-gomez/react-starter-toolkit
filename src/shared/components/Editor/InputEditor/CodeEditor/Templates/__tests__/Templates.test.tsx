import { render } from '@testing-library/react';
import Templates from '..';

const defaultProps = {
  submitTemplate: jest.fn(),
  onClearCodeEditor: jest.fn(),
};

describe('<Templates/>', () => {
  it('Should render Templates', () => {
    const { asFragment } = render(<Templates {...defaultProps} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
