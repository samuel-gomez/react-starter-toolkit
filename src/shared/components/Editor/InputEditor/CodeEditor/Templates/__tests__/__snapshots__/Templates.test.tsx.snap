// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Templates/> Should render Templates 1`] = `
<DocumentFragment>
  <div
    class="af-templates"
  >
    <div
      class="af-popover__wrapper"
    >
      <div
        class="af-popover__container af-popover__container--editor"
      >
        <div
          class="af-popover__container-over"
          role="presentation"
        >
          <button
            class="af-btn--circle af-btn--delete"
          >
            <i
              class="glyphicon glyphicon-trash"
            />
          </button>
        </div>
      </div>
    </div>
    <div
      class="af-popover__wrapper"
    >
      <div
        class="af-popover__container af-popover__container--editor"
      >
        <div
          class="af-popover__container-over"
          role="presentation"
        >
          <button
            class="af-btn--circle"
            id="add-list"
          >
            <i
              class="glyphicon glyphicon-list"
            />
          </button>
        </div>
      </div>
    </div>
    <div
      class="af-popover__wrapper"
    >
      <div
        class="af-popover__container af-popover__container--editor"
      >
        <div
          class="af-popover__container-over"
          role="presentation"
        >
          <button
            class="af-btn--circle"
            id="add-title__h1"
          >
            H1
          </button>
        </div>
      </div>
    </div>
    <div
      class="af-popover__wrapper"
    >
      <div
        class="af-popover__container af-popover__container--editor"
      >
        <div
          class="af-popover__container-over"
          role="presentation"
        >
          <button
            class="af-btn--circle"
            id="add-title__h2"
          >
            H2
          </button>
        </div>
      </div>
    </div>
    <div
      class="af-popover__wrapper"
    >
      <div
        class="af-popover__container af-popover__container--editor"
      >
        <div
          class="af-popover__container-over"
          role="presentation"
        >
          <button
            class="af-btn--circle"
            id="add-title__h3"
          >
            H3
          </button>
        </div>
      </div>
    </div>
    <div
      class="af-popover__wrapper"
    >
      <div
        class="af-popover__container af-popover__container--editor"
      >
        <div
          class="af-popover__container-over"
          role="presentation"
        >
          <button
            class="af-btn--circle"
            id="add-title__h4"
          >
            H4
          </button>
        </div>
      </div>
    </div>
    <div
      class="af-popover__wrapper"
    >
      <div
        class="af-popover__container af-popover__container--editor"
      >
        <div
          class="af-popover__container-over"
          role="presentation"
        >
          <button
            class="af-btn--circle"
            id="add-text"
          >
            P
          </button>
        </div>
      </div>
    </div>
    <div
      class="af-popover__wrapper"
    >
      <div
        class="af-popover__container af-popover__container--editor"
      >
        <div
          class="af-popover__container-over"
          role="presentation"
        >
          <button
            class="af-btn--circle"
            id="add-restitution__twoColumns"
          >
            <i
              class="glyphicon glyphicon-list-alt"
            />
          </button>
        </div>
      </div>
    </div>
    <div
      class="af-popover__wrapper"
    >
      <div
        class="af-popover__container af-popover__container--editor"
      >
        <div
          class="af-popover__container-over"
          role="presentation"
        >
          <button
            class="af-btn--circle"
            id="add-restitution__oneColumn"
          >
            <i
              class="glyphicon glyphicon-list-alt"
            />
          </button>
        </div>
      </div>
    </div>
    <div
      class="af-popover__wrapper"
    >
      <div
        class="af-popover__container af-popover__container--editor"
      >
        <div
          class="af-popover__container-over"
          role="presentation"
        >
          <button
            class="af-btn--circle"
            id="add-modal"
          >
            <i
              class="glyphicon glyphicon-modal-window"
            />
          </button>
        </div>
      </div>
    </div>
    <div
      class="af-popover__wrapper"
    >
      <div
        class="af-popover__container af-popover__container--editor"
      >
        <div
          class="af-popover__container-over"
          role="presentation"
        >
          <button
            class="af-btn--circle"
            id="add-panel__basic"
          >
            <i
              class="glyphicon glyphicon-credit-card"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
